src/ngraph_deassign_clusters.cc

```diff
@@ -78,36 +82,65 @@ static void MaybeLogPlacement(const Graph* graph) {
   int perc_assigned_clusters_of_total =
       (int)((nodes_assigned_a_cluster * 100.0) / number_of_nodes);
   int perc_assigned_clusters_of_marked =
-      nodes_marked_for_clustering > 0
+      num_nodes_marked_before_deassign > 0
           ? (int)((nodes_assigned_a_cluster * 100.0) /
-                  nodes_marked_for_clustering)
+                  num_nodes_marked_before_deassign)
           : 0;
 
-  std::cout << "Number of nodes in the graph: " << number_of_nodes << std::endl;
-  std::cout << "Number of nodes marked for clustering: "
-            << nodes_marked_for_clustering << " ("
+  std::cout << "\n";  // insert a new line at the start of NGTF_SUMMARY
+  std::cout << "NGTF_SUMMARY: Number of nodes in the graph: " << number_of_nodes
+            << std::endl;
+  // print out the number of nodes marked before deassign
+  std::cout << "NGTF_SUMMARY: Number of nodes marked for clustering: "
+            << num_nodes_marked_before_deassign << " ("
             << perc_marked_for_clustering_of_total << "% of total nodes)"
             << std::endl;
-  std::cout << "Number of nodes assigned a cluster: "
+  // print out the number of nodes that are running on NGraph after deassign
+  std::cout << "NGTF_SUMMARY: Number of nodes assigned a cluster: "
             << nodes_assigned_a_cluster << " ("
             << perc_assigned_clusters_of_total << "% of total nodes) \t"
             << " (" << perc_assigned_clusters_of_marked
             << "% of nodes marked for clustering) \t" << std::endl;
-  std::cout << "Number of ngraph clusters :" << final_cluster_map.size() - 1
-            << std::endl;
+  std::cout << "NGTF_SUMMARY: Number of ngraph clusters :"
+            << final_cluster_map.size() - 1 << std::endl;
 
   for (auto kv : final_cluster_map) {
     int cluster_idx = kv.first;
     if (cluster_idx != -1) {
-      std::cout << "Size of nGraph Cluster[" << cluster_idx << "]\t"
-                << kv.second.size() << std::endl;
+      std::cout << "NGTF_SUMMARY: Size of nGraph Cluster[" << cluster_idx
+                << "]:\t" << kv.second.size() << std::endl;
     }
   }
 
+  // log the ops gets deassigned
+  int deassigned_histogram_counter = 0;
+  int deassigned_histogram_size = deassigned_histogram.size();
+  std::cout << "NGTF_SUMMARY: Op_deassigned: ";
+  if (deassigned_histogram_size == 0) {
+    std::cout << "None" << std::endl;
+  } else {
+    // Descending sort
+    vector<std::pair<string, int>> v(begin(deassigned_histogram),
+                                     end(deassigned_histogram));
+    sort(begin(v), end(v),
+         [](const pair<string, int>& a, const pair<string, int>& b) {
+           return a.second > b.second;
+         });
+
+    for (auto node : v) {
+      if (++deassigned_histogram_counter < deassigned_histogram_size) {
+        std::cout << " " << node.first << " -> " << node.second << ",";
+      } else {
+        std::cout << " " << node.first << " -> " << node.second << std::endl;
+      }
+    }
```

Comment:
Minor:

I think the `if-else` is to avoid the trailing comma in the end.

`ng::join` can be used here: https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/util.hpp#L49

```
vector<string> string_vect(v.size());
transform(v.begin(), v.end(), string_vect.begin(), [](std::pair<string, int>& p){return p.first + " -> " + p.second});
std::cout << ng::join(string_vect)
```

or, some code duplication can be avoided by:
```
for (auto node : v) {
    bool endelem = node != v.back();
    std::cout << " " << node.first << " -> " << node.second << (endelem ? "" : ",");
}
```

Reply:
Yes, indeed to get rid of the comma in the end. 

Thanks for the suggestion, I will make the change based on the suggested method.